core/src/main/java/org/elasticsearch/common/lucene/search/function/WeightFactorFunction.java

```diff
@@ -106,5 +118,10 @@ public Explanation explainScore(int docId, Explanation subQueryScore) {
         public boolean needsScores() {
             return false;
         }
+
+        @Override
+        protected boolean doEquals(ScoreFunction other) {
```

Comment:
already done in the superclass for all of the functions. This specific function doesn't have any member field, so nothing more to do. Can add a comment.